### ISSUE TITLE
[Issue] could not resolve FW rule

### ISSUE THREAD

**Issue by roniyako** [GeoffWilliams/puppet-windows_firewall#10]
Error when doing a puppet run, below is the output

Windows 2016 

```
PS C:\Windows\system32> puppet agent -t
Info: Using configured environment 'production'
Info: Retrieving pluginfacts
Info: Retrieving plugin
Info: Retrieving locales
Info: Loading facts
Info: Caching catalog for Server
Error: /Stage[main]/Windows_fw::Enablefw/Resources[windows_firewall_rule]: Failed to generate additional resources using
 'generate': Execution of 'powershell.exe -ExecutionPolicy Bypass -File C:/ProgramData/PuppetLabs/puppet/cache/lib/ps/wi
ndows_firewall/ps-bridge.ps1 show' returned 1: could not resolve @{Microsoft.Windows.Apprep.ChxApp_1000.14393.2580.0_neu
tral_neutral_cw5n1h2txyewy?ms-resource://Micro
soft.Windows.Apprep.ChxApp/resources/DisplayName} in registry under HKCR:\Local Settings\MrtCache
At C:\ProgramData\PuppetLabs\puppet\cache\lib\ps\windows_firewall\ps-bridge.ps1:97 char:13
+             throw "could not resolve $($ref) in registry under $($sea ...
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (could not resol...ttings\MrtCache:String) [], RuntimeException
    + FullyQualifiedErrorId : could not resolve @{Microsoft.Windows.Apprep.ChxApp_1000.14393.2580.0_neutral_neutral_cw
   5n1h2txyewy?ms-resource://Microsoft.Windows.Apprep.ChxApp/resources/DisplayName} in registry under HKCR:\Local Set
  tings\MrtCache
Info: Applying configuration version puppetmaster-production-92d4a5fa3bf'
Error: Failed to apply catalog: Execution of 'powershell.exe -ExecutionPolicy Bypass -File C:/ProgramData/PuppetLabs/pup
pet/cache/lib/ps/windows_firewall/ps-bridge.ps1 show' returned 1: could not resolve @{Microsoft.Windows.Apprep.ChxApp_10
00.14393.2580.0_neutral_neutral_cw5n1h2txyewy?ms-resource://Micro
soft.Windows.Apprep.ChxApp/resources/DisplayName} in registry under HKCR:\Local Settings\MrtCache
At C:\ProgramData\PuppetLabs\puppet\cache\lib\ps\windows_firewall\ps-bridge.ps1:97 char:13
+             throw "could not resolve $($ref) in registry under $($sea ...
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (could not resol...ttings\MrtCache:String) [], RuntimeException
    + FullyQualifiedErrorId : could not resolve @{Microsoft.Windows.Apprep.ChxApp_1000.14393.2580.0_neutral_neutral_cw
   5n1h2txyewy?ms-resource://Microsoft.Windows.Apprep.ChxApp/resources/DisplayName} in registry under HKCR:\Local Set
  tings\MrtCache
```

**Comment by GeoffWilliams** [comment]
Ok this is failing to resolve a reference `@{Microsoft.Windows.Apprep.ChxApp_1000.14393.2580.0_neutral_neutral_cw5n1h2txyewy?ms-resource://Microsoft.Windows.Apprep.ChxApp/resources/DisplayName}` to a name by looking in `HKCR:\Local Settings\MrtCache`

For speed, I've been restricting name resolution to this search path but its likely we need to broaden the search.

Please could you search the windows registry on affected node for the string `@{Microsoft.Windows.Apprep.ChxApp_1000.14393.2580.0_neutral_neutral_cw5n1h2txyewy?ms-resource://Microsoft.Windows.Apprep.ChxApp/resources/DisplayName}` - this should take you to a value `DisplayName`.  Once you've found it please update this ticket with the windows registry search path it was found at and also what the DisplayName is - I'm curious as to why I didn't get the same problem. Perhaps it happens when you install additional Windows components.

If you can't find a value for this anywhere, then either your registry is corrupt or there is some additional magic required to resolve a name from the lookup key - any ideas?

**Comment by roniyako** [comment]
I found it in the following area.

HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Services\SharedAccess\Parameters\FirewallPolicy\FirewallRules

{6D084237-9A30-4B24-9F6D-448F3E4DDFBA}

v2.26|Action=Allow|Active=TRUE|Dir=Out|Profile=Domain|Profile=Private|Profile=Public|Name=@{Microsoft.Windows.Apprep.ChxApp_1000.14393.2580.0_neutral_neutral_cw5n1h2txyewy?ms-resource://Microsoft.Windows.Apprep.ChxApp/resources/DisplayName}|Desc=@{Microsoft.Windows.Apprep.ChxApp_1000.14393.2580.0_neutral_neutral_cw5n1h2txyewy?ms-resource://Microsoft.Windows.Apprep.ChxApp/resources/DisplayName}|LUOwn=S-1-5-21-1030466146-565472683-1424130068-500|AppPkgId=S-1-15-2-1255970798-2717750985-493741290-1721212560-3530798636-1829112236-3118580706|EmbedCtxt=@{Microsoft.Windows.Apprep.ChxApp_1000.14393.2580.0_neutral_neutral_cw5n1h2txyewy?ms-resource://Microsoft.Windows.Apprep.ChxApp/resources/DisplayName}|Platform=2:6:2|Platform2=GTEQ|

**Comment by GeoffWilliams** [comment]
fixed in 0.2.4
